### PR TITLE
[SYNPY-1362] High level best practices for project structure

### PR DESCRIPTION
**Pull request by thomasyu888**
**Problem**

There is currently not a best practices doc on project structure with your data.

### CODE REVIEW

File: docs/explanations/structuring_your_project.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+We recommend creating Synapse Teams for permission management on your Synapse project.  These are the recommended teams to create:
+
+* **<project> Admin** - This team should have "Administrator" access to the project.  This team should be used to manage the project and grant access to other teams.
+
+* **<project> Users** - Optional: If you want to grant all registered users on Synapse download access to your project, you can click the "Make Public" button in project's sharing settings.  This team should have "Can Download" access to the project.  This team should be used to grant download access to the project.
+
+Below are some key permission criterias to consider when setting up your project:
+
+* Create a Synapse Team and Project per data contributor if there are (1) multiple data contributors and (2) the data contributors should not have access to each other's raw data. You would then create a central "public" project that will contain the harmonized data. You can technically leverage local share settings by creating private folders in one project but managing local share settings is more complicated and not recommended.
```
[the remark this answers is not in the document]
Comment (BryanFauble) [reply docs/explanations/structuring_your_project.md:17]: Or maybe we have a "Read more" section at the bottom of the doc?

File: docs/explanations/structuring_your_project.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+When organizing your data for upload we have a preferred organization (flattened data layout) and an alternate option (hierarchy data layout) if your project requires that.
+
+> NOTE: If you and your contributing site decide to use a hierarchical file structure within your cloud storage location, please remember that each top-level folder and all of its subfolders must contain data of the same type (see details below).
+
+
+### Top Level Folder Names
+
+Top level folders correspond to the datasets being submitted. See the examples below. You can name your datasets in a way that is descriptive for your contributing site.
+
+You can use either the Hierarchy or Flattened data layout according to the examples below. In the hierarchical case, you would fill in one manifest and include all files in experiment/batches; in the flattened case, you would fill in one manifest for each top level folder.  The manifest would be Synapse annotations which can be used to query the data when a File View is created.
```
Comment (BryanFauble) [inline docs/explanations/structuring_your_project.md:34]:
```suggestion
You can use either the Hierarchy or Flattened data layout according to the examples below. In the **hierarchical case**, you would fill in one manifest and include all files in experiment/batches; in the **flattened case**, you would fill in one manifest for each top level folder.  The manifest would be Synapse annotations which can be used to query the data when a File View is created.
```

Comment (BryanFauble) [inline docs/explanations/structuring_your_project.md:34]: You reference a `manifest` here and in the file structure as CSV files. I know we have https://python-docs.synapse.org/explanations/manifest_tsv/ - which is a TSV file.

https://help.synapse.org/docs/Managing-Custom-Metadata-at-Scale.2004254976.html#ManagingCustomMetadataatScale-BatchUploadFileswithAnnotations as references this as a TSV.

Is CSV supported? Do we have documentation to link off to to explain what is contained within these files?

File: docs/explanations/structuring_your_project.md [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+```
+
+
+### To Create a Project Fileview with scope set to the project:
+
+* Add column contentType to the Fileview schema (default parameters for the column schema will work).
+* Give every Team Download level access to this fileview.
+* Note: creating this file view will not be possible if files/folders don’t yet exist in the center-specific projects; Synapse will not allow you to create a file view with an empty scope.
+* Make sure to add both file and folder entities to the scope of the Fileview.
+
```
Comment (BryanFauble) [inline docs/explanations/structuring_your_project.md:90]:
```suggestion

Read me about File Views [here](https://help.synapse.org/docs/Views.2011070739.html#Views-CreatingaFileView)
```

**Comment by BryanFauble** [comment]
@thomasyu888 You'll need to add this to the navigation in `mkdocs.yml` 

https://github.com/Sage-Bionetworks/synapsePythonClient/blob/develop/mkdocs.yml#L16

File: mkdocs.yml [hunk 4]
```diff
@@ -56,6 +56,7 @@ nav:
     - Properties vs Annotations: explanations/properties_vs_annotations.md
     - Manifest TSV: explanations/manifest_tsv.md
     - Benchmarking: explanations/benchmarking.md
+    - Structing Your Project: explanations/structuring_your_project.md
```
Comment (BryanFauble) [inline mkdocs.yml:59]:
```suggestion
    - Structuring Your Project: explanations/structuring_your_project.md
```

**Review by BryanFauble — APPROVED** [review]
Excellent work! LGTM!

File: docs/explanations/structuring_your_project.md [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+> Note: This page is a work in progress and will contain code examples at a later date.
+
+## Permissions Management
+
+We recommend creating Synapse Teams for permission management on your Synapse project so that you manage users that are in these teams instead of granting individual users access to your project.  These are the recommended teams to create:
+
+* **<project> Admin** - This team should have "Administrator" access to the project and is used to used to manage the project and grant access to other teams.
+* **<project> Users** - This team is optional, but can be used to grant a curated set of users download access to the project, by grant "Can Download" permission to the team. If you want to grant all registered users on Synapse download access to your project, click the "Make Public" button in project's sharing settings instead of creating and managing this team.
+
+Below are some key permission criterias to consider when setting up your project:
```
Comment (BWMac) [inline docs/explanations/structuring_your_project.md:14]:
```suggestion
Below are some key permission criteria to consider when setting up your project:
```

File: docs/explanations/structuring_your_project.md [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+
+* Create a Synapse Team and Project per data contributor if there are (1) multiple data contributors and (2) the data contributors should not have access to each other's raw data. You would then create a central "public" project that will contain the harmonized data. You can technically leverage [local share settings](https://help.synapse.org/docs/Sharing-Settings,-Permissions,-and-Conditions-for-Use.2024276030.html#SharingSettings,Permissions,andConditionsforUse-EditSharingSettingsonFiles,Folders,andTables) by creating private folders in one project but managing local share settings is more complicated and not recommended.
+* Do not mix data that requires different permission models within a folder. For example, if you have a folder that contains both public and private data, you should create two folders, one for public data and one for private data.  You can then grant the appropriate permissions to each folder. You can use local share settings to manage each file's permission, but this is not recommended!
```
Comment (BWMac) [inline docs/explanations/structuring_your_project.md:17]:
```suggestion
* Do not mix data that requires different permission models within a folder. For example, if you have a project that contains both public and private data, you should create two folders, one for public data and one for private data.  You can then grant the appropriate permissions to each folder. You can use local share settings to manage each file's permission, but this is not recommended!
```

File: docs/explanations/structuring_your_project.md [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+
+
+## Project Structure
+
+When organizing your data for upload we have a preferred organization (flattened data layout) and an alternate option (hierarchy data layout) if your project requires that. Synapse files are automaticlaly versioned when you create a file with the same filename, so be sure to account for that when organizing your folders.
```
Comment (BWMac) [inline docs/explanations/structuring_your_project.md:22]:
```suggestion
When organizing your data for upload we have a preferred organization (flattened data layout) and an alternate option (hierarchy data layout) if your project requires that. Synapse files are automatically versioned when you create a file with the same filename, so be sure to account for that when organizing your folders.
```

File: docs/explanations/structuring_your_project.md [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+You can use a view to:
+
+- Search and query many files, tables, projects, and submissions at once
+- View and edit file or table annotations in bulk
+- Group or link files, tables, projects, or submissions together by their annotations
+
+
+#### Creating the file view
+
+* Create a Fileview with the project set to the scope of the fileview
```
Comment (BWMac) [inline docs/explanations/structuring_your_project.md:96]: Some inconsistency here - "fileview", "file view", or "Fileview"

**Pull request merged**